errgroup.go

```diff
@@ -40,8 +40,10 @@ type Group struct {
 
 	wg sync.WaitGroup
 
-	errOnce sync.Once
```

Comment:
Since now we will be reading `err` continuously then we needed an `RWMutex` already.
We could keep the `sync.Once` but the same can be achieved with `RWMutex` that is already needed, so removed the `sync.Once`. Let me know if you think it's cleaner to keep both `sync.Once` and `sync.RWMutex`.